src/Razor/Microsoft.NET.Sdk.Razor/src/build/netstandard2.0/Microsoft.NET.Sdk.Razor.Components.Wasm.targets

```diff
@@ -25,6 +25,14 @@ Copyright (c) .NET Foundation. All rights reserved.
     <!-- Trimmer defaults -->
     <PublishTrimmed Condition="'$(PublishTrimmed)' == ''">true</PublishTrimmed>
     <TrimMode Condition="'$(TrimMode)' == ''">link</TrimMode>
+    <TrimmerRemoveSymbols Condition="'$(TrimmerRemoveSymbols)' == ''">false</TrimmerRemoveSymbols>
```

Comment:
The SDK defaults `TrimmerRemoveSymbols` based on `DebuggerSupport`. If `DebuggerSupport` is `false`, then the linker will remove the symbols. Since the Blazor SDK removes symbols from the publish output, I thought it was reasonable to always default `TrimmerRemoveSymbols=false` so even if DebuggerSupport is false in Release mode, we still get the linked symbols, in case the developer wants to archive the linked symbols.

cc @sbomer @vitek-karas

Reply:
Seems reasonable to me - I doubt that processing symbols significantly changes impacts link time. Curious that the logic to remove symbols from publish isn't shared.

Reply:
If the trimmer already has support for removing symbols, it might be useful to just piggy back on it. Would it help if we removed `BlazorWebAssemblyEnableDebugging` and defaulted `DebuggerSupport`  to `false`?

Reply:
> Would it help if we removed BlazorWebAssemblyEnableDebugging

I think this makes sense - might as well use the established properties in the core SDK and the runtime.

> and defaulted DebuggerSupport to false?

Is there ever a case where you'd want to preserve `.pdb`s for later debugging or perf sampling? I guess if users wanted symbols (but wanted to trim out the debug-only code for smaller apps) they could just set `TrimmerRemoveSymbols` themselves. I guess the current default behavior in Blazor WASM is to not get symbols. So I think that would be fine.

Reply:
In general I agree, only one concern:
* People will probably not realize that the build doesn't produce PDBs at the beginning
* They will only find out once they actually need the PDB for anything... but then it's too late as the deployed builds of the app won't have it
* Making it opt-in solves the problem only partially (for those who can redeploy and repro the issue)

I guess it depends on how we think people we use the PDBs for Blazor (and if they bring any value).

For example for server .NET Core apps (ASP.NET) I personally would definitely vote for always producing PDBs and just not bundling them to the final package.

Reply:
These are valid scenarios, but we've been removing pdbs by default during publish and haven't seen it be problematic. In general, I'd like to treat `dotnet publish` on blazor-wasm as a a deployment gesture and avoid having it complicated by adding Configuration as a second dimension. We always have the option of revisiting this decision if it comes to bite us.

Reply:
> I guess it depends on how we think people we use the PDBs for Blazor (and if they bring any value).

One place this brings value is when we start producing linker warnings. If the pdb is available, the linker will give source and line info in the warning. The developer will see the exact line of code causing the warning. That super useful IMO.

Reply:
My thinking is that having the .pdbs available somewhere - e.g. in the `obj` folder - by default, even if they are removed from the published assets, is useful. Honestly, I'm not sure why `dotnet publish` puts symbols in the publish folder by default.

https://github.com/dotnet/sdk/blob/44e5d1aa5ee51726096156f9c0d49e3b90c33f01/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets#L484

Reply:
> My thinking is that having the .pdbs available somewhere - e.g. in the obj folder - by default, even if they are removed from the published assets, is useful.

Those should still be around. We don't prevent creating the pdbs, just stop it from being copied to the publish output.

> If the pdb is available, the linker will give source and line info in the warning

Couldn't the linker still use pdbs and then discard these?

Reply:
> Couldn't the linker still use pdbs and then discard these?

If `TrimmerRemoveSymbols` is set to `true`, the linker won't update the symbols for the linked assemblies. Having the pre-trimmed symbols won't do you any good if you are using the trimmed assemblies - they won't match.

Reply:
Re source line info from linker: For this the linker needs the pdbs on the input - it doesn't matter if the linker itself will not write a pdb for this feature. That said I agree that having the pdbs (after linking) around somewhere is a good thing.